plugins/woocommerce/src/Internal/Admin/Settings.php

```diff
@@ -236,6 +239,13 @@ public function add_component_settings( $settings ) {
 
 		$settings['features'] = $this->get_features();
 
+		$data           = WCPayPromotionDataSourcePoller::get_instance()->get_specs_from_data_sources();
```

Comment:
The logic is somewhat duplicate of [getPromotions()](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/WCPayPromotion/Init.php#L124)

getPromotions() returns an array, so we probably need to add a method in the class to find woocommerce_payments:woopay

Reply:
Fixed on https://github.com/woocommerce/woocommerce/pull/39596/commits/c56335b70b8e5c4ab9685ac0af35ddfd7ea6c5e7.